### PR TITLE
1502 android app for data visualisation

### PR DESCRIPTION
**Pull request by sandradeng20**
BMSQueryApp that can monitor and control lab devices.

### CODE REVIEW

File: Apps/BMSQueryApp/app/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (sganguly-cares) [inline Apps/BMSQueryApp/app/proguard-rules.pro:1]: this file can be removed?

File: Apps/BMSQueryApp/app/src/main/assets/visualisation/visualisation.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        // change locale
+        for (i = 0; i < time.length; i++) {
+            time[i]['time'] = time[i]['time'].map(dateString => timeZoneConversion(dateString));
+        }
+
+        console.log(time);
+
+        // Plot data
+        timeseriesHandler.parseData(time);
+        // for (i = 0; i < timeseriesHandler._selectedData.length; i++) {
```
Comment (sganguly-cares) [inline Apps/BMSQueryApp/app/src/main/assets/visualisation/visualisation.js:36]: Remove unnecessary comments

File: Apps/BMSQueryApp/app/src/main/java/uk/ac/cam/cares/jps/bmsqueryapp/data/attribtue/EditableAttribute.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package uk.ac.cam.cares.jps.bmsqueryapp.data.attribtue;
```
Comment (sganguly-cares) [inline Apps/BMSQueryApp/app/src/main/java/uk/ac/cam/cares/jps/bmsqueryapp/data/attribtue/EditableAttribute.java:1]: typo "attribute" - fix if this does not involve a lot of major changes

File: Apps/BMSQueryApp/app/src/main/res/mipmap-anydpi-v33/ic_launcher.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (sganguly-cares) [inline Apps/BMSQueryApp/app/src/main/res/mipmap-anydpi-v33/ic_launcher.xml:1]: There are quite a few of the ic_launcher.xml files - are all of these needed?

File: Apps/BMSQueryApp/app/src/main/res/mipmap-mdpi/ic_cares_launcher_foreground.png
Comment (sganguly-cares) [inline Apps/BMSQueryApp/app/src/main/res/mipmap-mdpi/ic_cares_launcher_foreground.png:1]: Lot of these icon image files are present. Can you check if all are needed and if not, remove the unnecessary ones

**Review by sganguly-cares — CHANGES_REQUESTED** [review]
There are quite a few repeating pattern of files, not sure if all are needed - please remove if not.

File: Apps/BMSQueryApp/app/proguard-rules.pro (hunk 1 above)
Comment (sandradeng20) [reply Apps/BMSQueryApp/app/proguard-rules.pro:1]: This file cannot be removed. It is used to shrink down the size of the apk. Although there is nothing in this file, after removing it I will not be able to run the app with the IDE, so it is better to keep it.

Comment (sganguly-cares) [reply Apps/BMSQueryApp/app/proguard-rules.pro:1]: Okay, noted.

File: Apps/BMSQueryApp/app/src/main/res/mipmap-mdpi/ic_cares_launcher_foreground.png
Comment (sandradeng20) [reply Apps/BMSQueryApp/app/src/main/res/mipmap-mdpi/ic_cares_launcher_foreground.png:1]: These CARES icon images cannot be removed. They are for app icon under different screens and different icon sizes and shapes.

**Review by sganguly-cares — APPROVED** [review]

**Pull request merged**